src/libponyc/ast/lexer.c

```diff
@@ -555,7 +556,10 @@ static void normalise_string(lexer_t* lexer)
       }
       else
       {
-        if(ws_this_line < ws)
+        if(!isspace(c)) {
+          has_non_ws = true;
+        }
```

Comment:
You don't need the `{`/`}` braces for this - the prevailing style in this codebase is to omit braces when the body is just one statement.

```c
if(!isspace(c))
  has_non_ws = true;
```

Also note that if the braces were to be included (for a multi-line `if` statement), the prevailing style is to place the opening brace on its own line:


```c
if(!isspace(c))
{
  has_non_ws = true;
  some_other_statement();
}
```